packages/adapters/cache/src/lib/caches/transactions.ts

```diff
@@ -104,21 +115,57 @@ export class TransactionsCache extends Cache {
     });
   }
 
-  public async storeTxData(txs: XTransfer[]): Promise<void> {
-    for (const tx of txs) {
+  public async storeTransfers(transfers: XTransfer[]): Promise<void> {
+    const nonceIncreasesByDomain: { [domain: string]: boolean } = {};
+    const highestNonceByDomain: { [domain: string]: number } = {};
+    const newXCalls: { [transferId: string]: string } = {};
+    for (let transfer of transfers) {
+      const existing = await this.getTransferByTransferId(transfer.transferId);
+      // TODO: Sanity check: no update needed if this transfer is same as the one already stored!
+
+      // TODO: Sanity check: make sure that the existing transfer's metadata is the same as the old one!
+
+      // Update the existing transfer with the data from the new one; this will collate the transfer across
+      // domains, since our cache is indexed by transferId.
+      transfer = existing ? { ...existing, ...transfer } : transfer;
```

Comment:
What fields are in the `transfer`? would you want to always override existing fields with the newly pulled ones?

Reply:
see type `XTransfer` from utils

yes, pretty much always override... given that I add those two sanity checks above lol